back-end/src/controllers/verifyController.js

```diff
@@ -0,0 +1,40 @@
+const client = require('twilio')(process.env.ACCOUNT_SID, process.env.AUTH_TOKEN);
+ 
+ 
+exports.getCode = async (req, res) => {
+    console.log("******** getCode");
+    console.log(req.query.phonenumber);
+    console.log("channel: "+req.query.channel);
+    client
+        .verify
+        .services(process.env.VERIFY_SERVICE_SID)
+        .verifications
+        .create({
+            to: `+${req.query.phonenumber}`,
+            channel: req.query.channel
+        })
+        .then(data => {
+            console.log("res of getCode:"),
+            console.log(data),
+            res.status(200).send(data)
+        })
+};
+ 
+ 
+exports.verifyCode = async (req, res,err) => {
```

Comment:
This has a security flaw. A malicious user could create a denial of service. I think [twilio limits](https://www.twilio.com/docs/verify/api/rate-limits-and-timeouts) 60 requests in one minute, so a malicious user couldn't brute force crack a verification code, but a single malicious user could use all 60 requests forever. This should be rate limited. The back-end should check the request came from Slowvid - as a mitigation against a malicious user using `curl`.

Reply:
> The back-end should check the request came from Slowvid - as a mitigation against a malicious user using `curl`.

After researching, restricting the request came from Slowvid wouldn't be possible in our mock app. However a real app would restrict using [SafetyNet mobile attestation API](https://developer.android.com/training/safetynet/attestation).

My rate limit comment is still applicable.